src/shared/dissect-image.c

```diff
@@ -327,13 +328,20 @@ int dissect_image(
                 _cleanup_free_ char *name_stripped = NULL;
                 const char *full_path;
 
-                r = sd_device_get_sysattr_value(d, "loop/backing_file", &full_path);
-                if (r < 0)
-                        log_debug_errno(r, "Failed to lookup image name via loop device backing file sysattr, ignoring: %m");
-                else {
-                        r = raw_strip_suffixes(basename(full_path), &name_stripped);
+                if (original_filename) {
+                        /* The backing_file reference resolves symlinks but we need the original name for the sysext name check */
+                        r = raw_strip_suffixes(basename(original_filename), &name_stripped);
                         if (r < 0)
                                 return r;
+                } else {
+                        r = sd_device_get_sysattr_value(d, "loop/backing_file", &full_path);
+                        if (r < 0)
+                                log_debug_errno(r, "Failed to lookup image name via loop device backing file sysattr, ignoring: %m");
+                        else {
+                                r = raw_strip_suffixes(basename(full_path), &name_stripped);
```

Comment:
when you touch this, please replace use of basename() with path_extract_filename(), which we prefer these days.